### ISSUE TITLE
console assets fetch looks subject to races

### ISSUE THREAD

**Issue by davepacheco** [oxidecomputer/omicron#927]
I think there's some time-of-check-to-time-of-use issues with the way we serve console asset files.  In particular, if you had a/b/c/d, and the code got to "b" and confirmed it wasn't a symlink, and then you changed it to be a symlink to something completely different, I think we'd follow it to some/other/path/c/d.

**Comment by david-crespo** [comment]
I haven't dealt with something like this before. I'm guessing you can't lock down the files so they don't change, but rather make sure when you're pulling `a/b/c/d` that `a`, `a/b`, and `a/b/c` are not symlinks?

For future reference to the fixer, here is the relevant code:

https://github.com/oxidecomputer/omicron/blob/f91a12b658f48bba45f1ad17957173c069106480/nexus/src/external_api/console_api.rs#L358-L377

**Comment by davepacheco** [comment]
Yeah.  There's some discussion on oxidecomputer/dropshot#327.  I think this is basically why the `openat` family of functions exist -- you `open` the root, then `openat` the first directory component _within_ the root, `fstat` it to see if it's a symlink, then `openat` the next component, etc.  There's no opportunity for something you've checked whether it's a symlink to suddenly become something else because you're never referencing it (by name) twice.  You're always getting an fd, check that for being a symlink, then opening something relative to the _fd_ instead of an absolute path.
